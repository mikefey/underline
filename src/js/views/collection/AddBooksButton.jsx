import React from 'react';
import DropboxChooser from 'react-dropbox-chooser';
import config from 'config';
import style from './AddBooksButton.scss';


/**
 * Add books button component
 */
const AddBooksButton = (props) => {
  /**
   * Handler for successful download of file(s) from Dropbox
   * @param {Array<Object>} files - An array of objects with downloaded file data
   */
  const onDropboxFileSuccess = (files) => {
    if (files.length) {
      props.downloadBook(files[0].link);
    }
  };


  return (
    <div className={style.wrapper}>
      <DropboxChooser
        appKey={config.dropbox.appKey}
        success={files => onDropboxFileSuccess(files)}
        multiselect={false}
        extensions={['.epub']}
        linkType={'direct'}
      >
        <div className={style.icon} />
      </DropboxChooser>
    </div>
  );
};


/**
 * PropTypes
 * @prop {Function} downloadBook - function to dispatch a DOWNLOAD_BOOK action
 */
AddBooksButton.propTypes = {
  downloadBook: React.PropTypes.func.isRequired,
};

export default AddBooksButton;
