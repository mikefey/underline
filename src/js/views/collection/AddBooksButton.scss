@import '~colors.scss';

.wrapper {
  background-color: $rose;
  border-radius: 50%;
  cursor: pointer;
  height: 35px;
  position: absolute;
  right: 20px;
  top: 12px;
  width: 35px;
}

.icon {
  background-image: url('add-book-icon-white.png');
  background-size: cover;
  height: 18px;
  margin: 9px 0 0 10px;
  width: 18px;
}
