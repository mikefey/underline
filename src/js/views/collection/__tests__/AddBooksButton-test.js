import test from 'ava';
import React from 'react';
import { renderToString } from 'react-dom/server';

import AddBooksButton from './../AddBooksButton.jsx';

test('AddBooksButton component', (t) => {
  const collectionState = {
    collection: [],
  };
  const downloadBook = () => {
    return true;
  };

  const tree = renderToString(<AddBooksButton
    collectionState={collectionState}
    downloadBook={downloadBook}
  />);
  t.snapshot(tree);
});
