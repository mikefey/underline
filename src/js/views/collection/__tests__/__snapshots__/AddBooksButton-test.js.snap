// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddBooksButton component 1`] = `"<div class=\\"AddBooksButton__wrapper___3WnirUQZPEP4sp_\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"1687700633\\"><div data-reactid=\\"2\\"><div class=\\"AddBooksButton__icon___3Vo3Yz3c1Emxo_A\\" data-reactid=\\"3\\"></div></div></div>"`;
